service/tablet_allocator.cc

```diff
@@ -816,14 +842,21 @@ class load_balancer {
                 continue;
             }
             auto load = nodes[r.host].shards[r.shard].streaming_write_load;
-            if (load >= max_write_streaming_load) {
+            if ((load + load_delta) > max_write_streaming_load) {
                 lblogger.debug("Migration skipped because of write load limit on {} ({})", r, load);
                 return false;
             }
         }
         return true;
     }
 
+    // Precondition: all migration streaming info have same source and destination.
+    bool can_accept_load(node_load_map& nodes, const migration_streaming_info_vector& infos) {
+        // Since all migration info have source and destination, the load check can be easily done
```

Comment:
I thought about that, but we'd need to copy node_load_map, since we'd have to apply the load incrementally on each iteration, so that the next iteration will account for the previous one. I am not completely happy with this either, but I thought it was fine to have this precondition and improve it later.